pkg/liqonet/routing/common_test.go

```diff
@@ -0,0 +1,489 @@
+				}))
+			})
+		})
+
+		Context("when policy routing rule does not exist and we want to delete it", func() {
+			It("with destination net, should return false and nil", func() {
+				removed, err := delPolicyRoutingRule(dstNetCorrect, "", routingTableID)
+				Expect(removed).Should(Equal(false))
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("with source net, should return false and nil", func() {
+				removed, err := delPolicyRoutingRule("", srcNetCorrect, routingTableID)
+				Expect(removed).Should(Equal(false))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("when policy routing rule does exist and we want to delete it", func() {
+			JustBeforeEach(func() {
+				setUpRules()
+			})
+
+			JustAfterEach(func() {
+				tearDownRules()
+			})
+
+			It("with destination net, should return false and nil", func() {
+				removed, err := delPolicyRoutingRule("", existingRuleTo.Dst.String(), routingTableID)
+				Expect(removed).Should(Equal(true))
+				Expect(err).NotTo(HaveOccurred())
+				rule, err := getRule("", existingRuleTo.Dst.String(), routingTableID)
+				Expect(rule).To(BeNil())
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(Equal("rule not found"))
+			})
+
+			It("with source net, should return false and nil", func() {
+				removed, err := delPolicyRoutingRule(existingRuleFrom.Src.String(), "", routingTableID)
+				Expect(removed).Should(Equal(true))
+				Expect(err).NotTo(HaveOccurred())
+				rule, err := getRule("", existingRuleFrom.Src.String(), routingTableID)
+				Expect(rule).To(BeNil())
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(Equal("rule not found"))
+			})
+		})
+	})
+
+	Describe("flushing policy routing rules for a custom routing table", func() {
+		JustBeforeEach(func() {
+			setUpRules()
+		})
+
+		JustAfterEach(func() {
+			tearDownRules()
+		})
+
+		It("should remove all the policy routing rules referencing the custom table", func() {
+			err := flushRulesForRoutingTable(routingTableID)
+			Expect(err).NotTo(HaveOccurred())
+			// Check that no policy routing rules reference the custom routing table.
+			exists, err := existsRuleForRoutingTable(routingTableID)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(exists).Should(BeFalse())
+		})
+	})
+
+	Describe("getting interface index from which a given IP address is reachable", func() {
+
+		Context("when input parameters are not in the correct format", func() {
+			It("should return error on malformed IP address", func() {
+				index, err := getIFaceIndexForIP(gwIPWrong)
+				Expect(index).To(BeZero())
+				Expect(err).Should(Equal(&ParseIPError{Type: "IP address", IPToBeParsed: gwIPWrong}))
+			})
+		})
+
+		Context("when there is a route for the given IP address", func() {
+			It("should return nil and a correct link index", func() {
+				index, err := getIFaceIndexForIP(gwIPCorrect)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(index).Should(BeNumerically("==", dummylink1.Attrs().Index))
```

Comment:
Just a curiosity, why not a simple `Be(Equal(...))` instead of the `BeNumerically`?

Reply:
According to the [Gomega documentation](https://onsi.github.io/gomega/#benumericallycomparator-string-compareto-interface) it's the recommended way to make assertions with numerical types.

Reply:
👍 thanks